app/src/main/java/com/example/dineDelight/pages/HomeScreen.kt

```diff
@@ -0,0 +1,184 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import com.example.dineDelight.R
+import com.google.firebase.auth.FirebaseAuth
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun HomeScreen(navController: NavController) {
+    val firebaseAuth = FirebaseAuth.getInstance()
+    var currentUser by remember { mutableStateOf(firebaseAuth.currentUser) }
+    var showSignInDialog by remember { mutableStateOf(false) } // State to control dialog visibility
+
+    // Firebase Authentication state listener
+    DisposableEffect(Unit) {
+        val authStateListener = FirebaseAuth.AuthStateListener { auth ->
+            currentUser = auth.currentUser
+        }
+        firebaseAuth.addAuthStateListener(authStateListener)
+        onDispose {
+            firebaseAuth.removeAuthStateListener(authStateListener)
+        }
+    }
+
+    val bottomBarItems = listOf("Home", "Reservations", "Profile")
+
+    Scaffold(
+        topBar = {
+            SmallTopAppBar(
+                title = { Text("DineDelight") },
+                colors = TopAppBarDefaults.smallTopAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.primary,
+                    titleContentColor = MaterialTheme.colorScheme.onPrimary
+                )
+            )
+        },
+        bottomBar = {
+            NavigationBar {
+                bottomBarItems.forEach { item ->
+                    NavigationBarItem(
+                        selected = false,
+                        onClick = {
+                            when (item) {
+                                "Home" -> navController.navigate("home")
+                                "Reservations" -> navController.navigate("reservations")
+                                "Profile" -> navController.navigate("profile")
+                            }
+                        },
+                        icon = {
+                            Icon(
+                                painter = painterResource(id = R.drawable.dining),
+                                contentDescription = item
+                            )
+                        },
+                        label = { Text(item) }
+                    )
+                }
+            }
+        }
+    ) { padding ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(padding),
+            contentAlignment = Alignment.Center
+        ) {
+            Column(horizontalAlignment = Alignment.CenterHorizontally) {
+                if (currentUser != null) {
+                    // Show a personalized welcome message if the user is logged in
+                    Text(
+                        text = "Welcome back, ${currentUser?.email ?: "User"}!",
```

Comment:
When I logged in for the first time I got a "Welcome back" message, which was odd.
Instead, we could use a generic "Hello" message to resolve this issue.

```suggestion
                        text = "Hello, ${currentUser?.email ?: "User"}!",
```

Reply:
ok